File: Submissions/#0107_div2/A/main.java
import java.util.*;

public class Main {

  public static void main (String[] args) {

    Scanner sc = new Scanner(System.in);
    int n = sc.nextInt();
    int k = sc.nextInt();
    int l = sc.nextInt();
    int c = sc.nextInt();
    int d = sc.nextInt();
    int p = sc.nextInt();
    int nl = sc.nextInt();
    int np = sc.nextInt();
    sc.close();

    int answer = k * l / nl;
    answer = Math.min(answer, c * d);
    answer = Math.min(answer, p / np);

    System.out.println(answer / n);

  }

}

File: Submissions/#0107_div2/A/main.cpp
#include <bits/stdc++.h>
using namespace std;

int main() {

  int n, k, l, c, d, p, nl, np;
  cin >> n >> k >> l >> c >> d >> p >> nl >> np;

  int answer = k * l / nl;
  answer = min(answer, c * d);
  answer = min(answer, p / np);

  cout << answer / n << endl;

}